src/App.tsx

```diff
@@ -1,56 +1,62 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import { debounce } from 'lodash';
+import classNames from 'classnames';
 import './App.scss';
-import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isFocused, setIsFocused] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handelQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setIsFocused(true);
+    if (!event.target.value) {
+      setSelectedPerson(null);
+      setIsFocused(false);
+    }
```

Comment:
```suggestion
    const { value } = event.target;
    
    setQuery(value);
    applyQuery(value);
    setIsFocused(true);
    
    if (!event.target.value) {
      setSelectedPerson(null);
      setIsFocused(false);
    }
```
You can use destructing and don't repeat the same twice. Also, add blank lines to group operations for better readability